Core.DomainModel/Events/EntityBeingRemovedEvent.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Core.DomainModel.Shared;
+
+namespace Core.DomainModel.Events
+{
+    public class EntityBeingRemovedEvent<TEntity> : EntityLifeCycleEvent<TEntity>
```

Comment:
why not add context so it can be distinguished from the other one..

`UserBeingRemovedFromOrganizationEvent` and then not generic but hard coded to the user since this is a very special case